### PR TITLE
clean up vcd generation pass

### PR DESCRIPTION
**Pull request by jsn1993**

### CODE REVIEW

File: pymtl3/passes/VcdGenerationPass.py [hunk 1]
```diff
@@ -139,18 +87,12 @@ def dump_vcd():
 
     all_components = set()
 
-    # We only collect non-sliced leaf signals
+    # We only collect top level signals, and squash bitstruct into a
```
Comment (ptpan) [inline pymtl3/passes/VcdGenerationPass.py]: Missing something in comment?

File: pymtl3/passes/VcdGenerationPass.py [hunk 2]
```diff
@@ -201,12 +143,9 @@ def to_bits( obj ):
       if isinstance( obj, Bits ):
         return obj
       # BitStruct
-      bits = [to_bits(getattr(obj, v)) for v in obj.__bitstruct_fields__.keys()]
-      for _bits in bits[1:]:
-        bits[0] = (bits[0] << _bits.nbits) | _bits
-      return bits[0]
+      return concat( *[ to_bits(getattr(obj, v)) for v in obj.__bitstruct_fields__.keys() ] )
```
Comment (ptpan) [inline pymtl3/passes/VcdGenerationPass.py]: Should we include the to_bits function in pymtl or keep it a user space thing as it is now? I think Kai’s visualization pass needs something similar?

File: pymtl3/passes/VcdGenerationPass.py [hunk 3]
```diff
@@ -295,4 +234,56 @@ def recurse_models( m, level ):
     print( '\n#0\nb0b1 {}\n'.format( net_symbol_mapping[ vcdmeta.clock_net_idx ] ),
            file=vcdmeta.vcd_file, flush=True )
 
-    return make_dump_vcd(top, net_symbol_mapping, trimmed_value_nets, last_values)
+    # Given top, net_symbol_mapping, trimmed_value_nets, last_values, `make_dump_vcd`
```
Comment (ptpan) [inline pymtl3/passes/VcdGenerationPass.py]: The comment here is outdated.

File: pymtl3/passes/VcdGenerationPass.py (hunk 2 above)
Comment (jsn1993) [reply pymtl3/passes/VcdGenerationPass.py]: yes it does. we can add it to pymtl3.datatypes?

**Pull request merged**